src/gui/torrentcategorydialog.cpp

```diff
@@ -128,12 +135,62 @@ void TorrentCategoryDialog::setCategoryName(const QString &categoryName)
     m_ui->textCategoryName->setText(categoryName);
 }
 
-QString TorrentCategoryDialog::savePath() const
+BitTorrent::CategoryOptions TorrentCategoryDialog::categoryOptions() const
+{
+    BitTorrent::CategoryOptions categoryOptions;
+    categoryOptions.savePath = m_ui->comboSavePath->selectedPath();
+    if (m_ui->comboUseDownloadPath->currentIndex() == 1)
+        categoryOptions.downloadPath = {true, m_ui->comboDownloadPath->selectedPath()};
+    else if (m_ui->comboUseDownloadPath->currentIndex() == 2)
+        categoryOptions.downloadPath = {false};
+
+    return categoryOptions;
+}
+
+void TorrentCategoryDialog::setCategoryOptions(const BitTorrent::CategoryOptions &categoryOptions)
+{
+    m_ui->comboSavePath->setSelectedPath(categoryOptions.savePath);
+    if (categoryOptions.downloadPath)
+    {
+        m_ui->comboUseDownloadPath->setCurrentIndex(categoryOptions.downloadPath->enabled ? 1 : 2);
+        m_ui->comboDownloadPath->setSelectedPath(categoryOptions.downloadPath->enabled ? categoryOptions.downloadPath->path : QString());
+    }
+    else
+    {
+        m_ui->comboUseDownloadPath->setCurrentIndex(0);
+        m_ui->comboDownloadPath->setSelectedPath({});
+    }
+}
+
+void TorrentCategoryDialog::categoryNameChanged(const QString &categoryName)
 {
-    return m_ui->comboSavePath->selectedPath();
+    using namespace Utils::Fs;
```

Comment:
Seems it isn't frequently used so I would ditch it. (same in `TorrentCategoryDialog::useDownloadPathChanged`)

Reply:
> Seems it isn't frequently used so I would ditch it. (same in `TorrentCategoryDialog::useDownloadPathChanged`)

Ok.